persistent/Database/Persist/TH.hs

```diff
@@ -1185,19 +1200,21 @@ fieldError tableName fieldName err = mconcat
     ]
 
 mkEntity :: EntityMap -> MkPersistSettings -> EntityDef -> Q [Dec]
-mkEntity entityMap mps entDef = do
-    entityDefExp <-
-        if mpsGeneric mps
-           then liftAndFixKeys entityMap entDef
-           else makePersistEntityDefExp mps entityMap entDef
```

Comment:
I had to remove this optimization, since it wasn't handling `FieldDef` for `MigrationOnly` stuff.